### ISSUE TITLE
Change babel strategy (low priority: for HUGE production apps only)

### ISSUE THREAD

**Issue by mattkrick** [mattkrick/cashay#15]
Right now babel is used to write a function call that will create a normalization schema. This is useful for avoiding the large load of the clientSchema, but these function calls can be 10x bigger than the fragments they replace.
A very basic app has a clientSchema that is 1.5Kb gzipped (removed description, etc. fields from the introspectionQuery). Even if a production app were 10x bigger, it'd still be fine to ignore.

HOWEVER, if it's facebook-sized, a 100Kb clientSchema could exist. In this scenario, I could use babel + webpack.
Step 1: for each `type` in the clientSchema, write it to a .json where the name is type.name.value.
Step 2: In babel, parse the queryString into an AST. Make a list of all the types used. Map those to require statements where each one add itself to a prop of a `types` constant. 
Step 3: Reference the types constant directly.

This is superior to the current (relay) strategy in a few ways:
- it eliminates bloat of each babelfied TaggedTemplateString
- it eventually loads the entire clientSchema onto the client, loading each type only once
- by loading per-require, if the client needs the whole clientSchema (eg graphiql), it can request all requires and only receive the diff, reducing transmission
- webpack can handle where to split the requires per page

**Comment by mattkrick** [comment]
One thing that facebook does internally is replace the queryString with a short UID on the client. this eliminates over-the-wire transfer of a huge, declarative string. However, Cashay needs that string so it can parse the AST.

We could still have the queryString on the client & just send a UID for frequent queries, like how we do for subscriptions.
However, if we wanted to completely remove the queryString from the client, then we'd need the AST, which is what relay does. 

More thought needed....

**Comment by mattkrick** [comment]
Another possibility is combining the requestAST with the schema in babel.
Doing so provides, a single object to traverse instead of walking 2 at the same time. Computationally, this is a good thing.
The bad part is that if done naively, it'll be a larger payload, since we'll be including more info than necessary.

Ideally, we follow the original plan of sending the useful parts of the queries down to the client via webpack.
Then, when replacing the queryString with a reqAST, add in requires that will grab the necessary info from those useful schema parts.

eg Babel compilation step:
```
reqAST = parse(queryString);
reqAST.selectionSet.selections[0]._expectedArgs = require('./schemaParts/getPosts.json').args
reqAST.selectionSet.selections[0]._expectedFields = require('./schemaParts/getPosts.json').fields
```

**Comment by dustinfarris** [comment]
@mattkrick I think I am bumping against this logic.  Getting `Cannot read property 'selections' of null`.  I'm not sure where to start digging here; can you point me in the right direction?

**Comment by dustinfarris** [comment]
Created #131
